docs/examples/collision-particles.py

```diff
@@ -0,0 +1,164 @@
+    lines = [edge1, -edge1, edge2, edge3, edge4, edge5]
+    return lines
+
+
+##############################################################################
+# Here we define collision between walls-particles and particle-particle.
+# When collision happens, the particle with lower velocity gets the
+# color of the particle with higher velocity
+
+def collision():
+    global xyz
+    num_vertices = vertices.shape[0]
+    sec = np.int(num_vertices / num_particles)
+
+    for i, j in np.ndindex(num_particles, num_particles):
+
+        if (i == j):
+            continue
+        distance = np.linalg.norm(xyz[i] - xyz[j])
+        vel_mag_i = np.linalg.norm(vel[i])
+        vel_mag_j = np.linalg.norm(vel[j])
+        # Collision happens if the distance between the centars of two
+        # particles is less or equal to the sum of their radii
+        if (distance <= (radii[i] + radii[j])):
+            vel[i] = -vel[i]
+            vel[j] = -vel[j]
+            if vel_mag_j > vel_mag_i:
+                vcolors[i * sec: i * sec + sec] = \
+                    vcolors[j * sec: j * sec + sec]
+            if vel_mag_i > vel_mag_j:
+                vcolors[j * sec: j * sec + sec] = \
+                    vcolors[i * sec: i * sec + sec]
+            xyz[i] = xyz[i] + vel[i] * dt
+            xyz[j] = xyz[j] + vel[j] * dt
+    # Collision between particles-walls;
+    vel[:, 0] = np.where(((xyz[:, 0] <= - 0.5 * box_lx + radii[:]) |
+                          (xyz[:, 0] >= (0.5 * box_lx - radii[:]))),
+                         - vel[:, 0], vel[:, 0])
+    vel[:, 1] = np.where(((xyz[:, 1] <= - 0.5 * box_ly + radii[:]) |
+                          (xyz[:, 1] >= (0.5 * box_ly - radii[:]))),
+                         - vel[:, 1], vel[:, 1])
+    vel[:, 2] = np.where(((xyz[:, 2] <= -0.5 * box_lz + radii[:]) |
+                          (xyz[:, 2] >= (0.5 * box_lz - radii[:]))),
+                         - vel[:, 2], vel[:, 2])
+
+
+##############################################################################
+# We define position, velocity, color and radius randomly for 50 particles
+# inside the box.
+
+global xyz
+num_particles = 50
+box_lx = 20
+box_ly = 20
+box_lz = 10
+steps = 1000
+dt = 0.05
+xyz = np.array([box_lx, box_ly, box_lz]) * (np.random.rand(num_particles, 3)
+                                            - 0.5) * 0.6
+vel = 4 * (np.random.rand(num_particles, 3) - 0.5)
+colors = np.random.rand(num_particles, 3)
+radii = np.random.rand(num_particles) + 0.01
+
+##############################################################################
+# With box, streamtube and sphere actors, we can create the box, the
+# edges of the box and the spheres respectively.
+
+scene = window.Scene()
+box_centers = np.array([[0, 0, 0]])
+box_directions = np.array([[0, 1, 0]])
+box_colors = np.array([[255, 255, 255]])
+box_actor = actor.box(box_centers, box_directions, box_colors,
+                      scale=(box_lx, box_ly, box_lz))
+utils.opacity(box_actor, 0.2)
```

Comment:
this part will change as soon as #283 is merged. It fixed the bug you reported and you can do 

`box_colors = np.array([[1, 1, 1, 0.2]])`